src/Client.php

```diff
@@ -4,15 +4,25 @@
 
 use Aws\CognitoIdentityProvider\CognitoIdentityProviderClient;
 use Aws\Exception\AwsException;
-use Aws\Result;
-use Dvsa\Contracts\Auth\ClientException;
-use Dvsa\Contracts\Auth\InvalidTokenException;
+use Dvsa\Contracts\Auth\AccessTokenInterface;
+use Dvsa\Contracts\Auth\Exceptions\ChallengeException;
+use Dvsa\Contracts\Auth\Exceptions\ClientException;
+use Dvsa\Contracts\Auth\Exceptions\InvalidTokenException;
 use Dvsa\Contracts\Auth\OAuthClientInterface;
+use Dvsa\Contracts\Auth\ResourceOwnerInterface;
 use Firebase\JWT\JWK;
 use Firebase\JWT\JWT;
 
 class Client implements OAuthClientInterface
 {
+    /**
+     * When checking nbf, iat or expiration times on tokens, we want to provide
+     * some extra leeway time to account for clock skew.
+     *
+     * @var int
+     */
+    public static $leeway = 0;
```

Comment:
Do we need this if it's going to be 0? Or is the plan to figure out what it should be based on testing?

Reply:
Good question, _I think_ we need this to allow consumers to set their own desired leeway. In my opinion, there isn't a "correct" value for this as every scenario will be different depending on infrastructure and risk appetite.

Consumers can (recommended) to set this value while setting up the class in their app container (like [this](https://github.com/dvsa/authentication-examples/pull/10/files#diff-01c87be29887e0b4c02e0fd783f4c529c220d0aa34308ea01458aceaedabc526R206)). Using the skeleton app as an example of clock skew, as we are usually serving the app locally, the time on our local machine and AWS _could_ differ on some occasions (my local environment was 2 seconds behind AWS yesterday).

Reply:
Ah yes the consumes would be be wanting to set it. Had overlooked that.